lib/http.ts

```diff
@@ -0,0 +1,88 @@
+import {ChainInfo, NetworkClient, RandomnessBeacon} from './drand'
+import Chain from './chain'
+import {controllerWithParent} from './abort'
+import PollingWatcher from './polling-watcher'
+
+export default class HTTP implements NetworkClient {
+    private readonly watcher: Watcher
+    private controllers: Array<AbortController>
+
+    constructor(
+        private readonly url: string,
+        private readonly chainInfo: ChainInfo) {
+        this.watcher = new PollingWatcher(this, chainInfo)
+        this.controllers = []
+    }
+
+    async get(round = 0, options: ClientOptions) {
+        options = options || {}
+
+        const controller = controllerWithParent(options.signal)
+        this.controllers.push(controller)
+
+        try {
+            const url = `${this.url}/public/${round || 'latest'}${options.noCache ? '?' + Date.now() : ''}`
```

Comment:
are you handling the chain hash at the url level then?

Reply:
I just translated the old impl and curiously it ignores the `chainHash` and just calls the default `public`, e.g. `https://pl-us.testnet.drand.sh/public/latest`.

This needs rethought for sure

Reply:
ahhhh right, that makes sense then.  In that case, I will raise a separate PR for that when I implement unchained beacon verification (which I have already basically done in timelock, so I will port it over)